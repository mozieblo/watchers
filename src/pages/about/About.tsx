import React, { FC } from 'react';
import { useIntl } from 'react-intl';
import styled from 'styled-components';

interface IWrapper {
    primary?: boolean;
}

const About: FC = () => {
    const intl = useIntl();
    return (
        <>
            <Wrapper primary>{intl.formatMessage({ id: 'about.content' })}</Wrapper>
            <Wrapper>{intl.formatMessage({ id: 'about.content' })}</Wrapper>
        </>
    );
};

const Wrapper = styled.section<IWrapper>`
    color: ${({ theme }) => theme.primaryColor};
    background-color: ${props => (props.primary ? 'yellow' : 'black')};
    border: 1px solid violet;
`;

export default About;
