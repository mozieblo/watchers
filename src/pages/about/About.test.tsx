import React from 'react';
import { render } from '@testing-library/react';
import { intlWrapper } from '../../config/test/intlWrapper';
import About from './About';

test('About: should render some text', () => {
    const { getAllByText } = render(<About />, { wrapper: intlWrapper });

    expect(getAllByText('About page.'));
    expect(getAllByText).toMatchSnapshot();
});
