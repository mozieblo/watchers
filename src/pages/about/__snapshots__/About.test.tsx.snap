// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About: should render some text 1`] = `[Function]`;
